server/src/test/java/org/folio/reservoir/solr/impl/VertxSolrClientTest.java

```diff
@@ -0,0 +1,212 @@
+package org.folio.reservoir.solr.impl;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.json.DecodeException;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.handler.BodyHandler;
+import org.apache.solr.common.SolrDocumentList;
+import org.apache.solr.common.SolrInputDocument;
+import org.apache.solr.common.params.MapSolrParams;
+import org.folio.reservoir.solr.VertxSolrClient;
+import org.junit.AfterClass;
+import org.junit.Assume;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+@RunWith(VertxUnitRunner.class)
+public class VertxSolrClientTest {
+  private static Vertx vertx;
+
+  private static int mockPort = 9230;
+
+  private static String solrUrl = "http://localhost:8983/solr";
+  private static String mockUrl = "http://localhost:" + mockPort + "/solr";
+
+  private final static String COLLECTION = "col1";
+
+  private static WebClient webClient;
+
+  private static boolean hasSolr;
+
+  @BeforeClass
+  public static void beforeClass(TestContext context) {
+    vertx = Vertx.vertx();
+    webClient = WebClient.create(vertx);
+    checkSolr()
+        .onSuccess(x -> hasSolr = true)
+        .recover(x -> Future.succeededFuture())
+        .compose(x -> setupMockSolr())
+        .onComplete(context.asyncAssertSuccess());
+  }
+
+  static Future<Void> checkSolr() {
+    return webClient.getAbs(solrUrl + "/" + COLLECTION + "/select")
+        .addQueryParam("q", "title:title1")
+        .addQueryParam("wt", "json")
+        .send()
+        .map(res -> {
+          JsonObject ret = res.bodyAsJsonObject();
+          if (!ret.containsKey("responseHeader")) {
+            throw new DecodeException("Does not have responseHeader");
+          }
+          return null;
+        });
+  }
+
+  static Future<Void> setupMockSolr() {
```

Comment:
There seems to be a solr test contaier: https://www.testcontainers.org/modules/solr/

Reply:
Exactly, I'm going to use that.